### PR TITLE
Use Map.TryGetValue instead of Map.TryFind

### PR DESCRIPTION
**Pull request by forki**
replaces #4839

### CODE REVIEW

**Comment by forki** [comment]
@kevinransom this one is ready

**Comment by KevinRansom** [comment]
@forki, conflicts.

**Comment by forki** [comment]
Lol @dsyme just merged yet another cleanup. This is really hard right now

Kevin Ransom (msft) <notifications@github.com> schrieb am Mo., 11. März
2019, 21:54:

**Comment by KevinRansom** [comment]
@forki, I'm sorry, he's not supposed to do that.  Sometimes he seems to forget ...

**Comment by forki** [comment]
No it's fine. Cleanups should go in fast so that other PRs have it easier.
It's just that it hit my PRs multiple times here 😂

Kevin Ransom (msft) <notifications@github.com> schrieb am Mo., 11. März
2019, 21:59:

File: tests/EndToEndBuildTests/ProvidedTypes/ProvidedTypes.fs [hunk 1]
```diff
@@ -48,9 +48,6 @@ namespace ProviderImplementation.ProvidedTypes
             let toObj x = match x with UNone -> null | USome x -> x
             let ofObj x = match x with null -> UNone | x -> USome x
 
-
-        let tryFindMulti k map = match Map.tryFind k map with Some res -> res | None -> [| |]
-
         let splitNameAt (nm:string) idx =
```
Comment (KevinRansom) [inline tests/EndToEndBuildTests/ProvidedTypes/ProvidedTypes.fs]: not this.  This file came from the type provider sdk, so we shouldn't change it.  You might submit a change to the type provider sdk though.

File: src/absil/il.fs [hunk 2]
```diff
@@ -3169,10 +3169,13 @@ let mkILNestedExportedTypesLazy (l:Lazy<_>) =
 
 let mkILResources l =  ILResources l
 
-let addMethodImplToTable y tab =
+let addMethodImplToTable y (tab:Map<_,_>) =
     let key = (y.Overrides.MethodRef.Name, y.Overrides.MethodRef.ArgTypes.Length)
-    let prev = Map.tryFindMulti key tab
-    Map.add key (y::prev) tab
+    match tab.TryGetValue key with
```
Comment (dsyme) [inline src/absil/il.fs]: I'd rather not make this kind of change.  The declarative `tryFindMulti` has been expanded out into larger code, it's harder to read and think about.  The adjustment should be in tryFindMulti.

File: src/absil/illib.fs [hunk 3]
```diff
@@ -667,9 +667,6 @@ type Lock<'LockTokenType when 'LockTokenType :> LockToken>() =
 /// Get an initialization hole 
 let getHole r = match !r with None -> failwith "getHole" | Some x -> x
 
-module Map =
```
Comment (dsyme) [inline src/absil/illib.fs]: Please revert

File: src/fsharp/NameResolution.fs [hunk 4]
```diff
@@ -734,11 +734,13 @@ let AddExceptionDeclsToNameEnv bulkAddMode nenv (ecref:TyconRef) =
 
 /// Add a module abbreviation to the name resolution environment
 let AddModuleAbbrevToNameEnv (id:Ident) nenv modrefs =
-    {nenv with
+    { nenv with
        eModulesAndNamespaces =
-         let add old nw = nw @ old
-         NameMap.layerAdditive add (Map.add id.idText modrefs Map.empty) nenv.eModulesAndNamespaces }
```
Comment (dsyme) [inline src/fsharp/NameResolution.fs]: Please revert - why expand out `layerAdditive`??

File: src/fsharp/NameResolution.fs [hunk 5]
```diff
@@ -755,12 +757,19 @@ let rec AddModuleOrNamespaceRefsToNameEnv g amap m root ad nenv (modrefs: Module
     if isNil modrefs then nenv else
     let modrefsMap = modrefs |> NameMap.ofKeyedList (fun modref -> modref.DemangledModuleOrNamespaceName)
     let addModrefs tab =
-         let add old nw =
-             if IsEntityAccessible amap m ad nw then
-                 nw :: old
-             else
-                 old
-         NameMap.layerAdditive add modrefsMap tab
```
Comment (dsyme) [inline src/fsharp/NameResolution.fs]: Likewise, please revert

**Comment by dsyme** [comment]
I looked over this, I'm not sure it's really an improvement?

**Comment by dsyme** [comment]
@forki Sorry about that :) thanks in any case!!

**Pull request closed without merging**